org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -875,6 +878,18 @@ public void didChangeWorkspaceFolders(DidChangeWorkspaceFoldersParams params) {
 		return computeAsync((monitor) -> GetRefactorEditHandler.getEditsForRefactor(params));
 	}
 
+	@Override
+	public CompletableFuture<MoveDestinationsResponse> getMoveDestinations(MoveDestinationsParams params) {
+		logInfo(">> java/getMoveDestinations");
+		return computeAsync((monitor) -> MoveHandler.getMoveDestinations(params));
+	}
+
+	@Override
+	public CompletableFuture<RefactorWorkspaceEdit> moveFile(MoveFileParams params) {
```

Comment:
I think we should rename it to moveResources(MoveResourcesParams params), since we might handle package moves in the future

Reply:
A pretty good suggestion, thanks!

But i prefer to make the api more generic `move(MoveParams params)`, actually we will have more move scenarios. I'm also working on support moveInstanceMethod, moveStaticMember, moveTypeToNewFile, moveTypeToOtherClass.